### DISCUSSION TITLE
bind docker container to host ip

### DISCUSSION THREAD

**Discussion by aperrot42** [lima-vm/lima#1379] [Q&A]
Try to replace docker desktop by lima for local dev with nomad

I am using lima docker-rootful template

I am trying to run a docker image and bind to my host ip :
```
➜  docker run --rm -p192.168.1.120:28033:80 nginx # dockerd on lima complains
docker: Error response from daemon: driver failed programming external connectivity on endpoint magical_jemison (87d8617ca846be705c03b628c90a8bc2aba45378d4c494badc48b6f3ed3c6070): Error starting userland proxy: listen tcp4 192.168.1.120:28033: bind: cannot assign requested address.
```

I am trying to do that to dev with a Nomad scheduler on my mac m1 (I have no control over the docker command, dockerd communication is managed by a nomad built-in plugin).

Surprisingly, this works with docker desktop on mac :
```
➜ docker run --rm -p192.168.1.120:28033:80 nginx # docker desktop does not complain...
```

How can I "trick" dockerd in lima into thinking that it can bind to my host ip ?

**Comment by afbjorklund** [comment]
* https://github.com/lima-vm/lima/discussions/1187

**Reply by aperrot42** [reply]
Thanks @afbjorklund gave lima another try tonight, it seems like it does not work :
I created a file in `$HOME/.lima/_config/override.yaml` with this content :

```yaml
portForwards:
- guestIP: "127.0.0.1"
  guestPortRange: [1, 65535]
  hostIP: "0.0.0.0"
  hostPortRange: [1, 65535]
```

Stopped and restarted lima vm.

still no luck :

```
➜  ~ docker run --rm -p192.168.1.120:28033:80 nginx 
docker: Error response from daemon: driver failed programming external connectivity on endpoint jolly_hawking (3188a7c314a73d57792599252f222f2141f06bcd51b7ffcd3422fb409db080b8): Error starting userland proxy: listen tcp4 192.168.1.120:28033: bind: cannot assign requested address.
```
